src/composeappmanager.cc

```diff
@@ -334,5 +334,21 @@ void ComposeAppManager::handleRemovedApps(const Uptane::Target& target) const {
 }
 
 std::string ComposeAppManager::getCurrentHash() const { return sysroot_->getCurDeploymentHash(); }
-
-std::string ComposeAppManager::runningApps() const { return app_engine_->runningApps(); }
+Json::Value ComposeAppManager::getRunningAppsInfo() const { return app_engine_->getRunningAppsInfo(); }
+std::string ComposeAppManager::getRunningAppsInfoForReport() const {
+  std::string result;
+  const auto apps_info = getRunningAppsInfo();
+  for (Json::ValueConstIterator ii = apps_info.begin(); ii != apps_info.end(); ++ii) {
+    result += ii.key().asString() + ": " + (*ii)["uri"].asString() + "; state: " + (*ii)["state"].asString() + "\n";
+    ;
+
+    Json::Value services = (*ii)["services"];
+    for (Json::ValueConstIterator jj = services.begin(); jj != services.end(); ++jj) {
+      result += "\t" + jj.key().asString() + ": " + (*jj)["hash"].asString();
+      result += "; image: " + (*jj)["image"].asString();
+      result += "; state: " + (*jj)["state"].asString();
+      result += "; status: " + (*jj)["status"].asString() + "\n";
```

Comment:
i worried status wouldn't be interesting, but it kind of is. this will let you see if things restarted that shouldn't have?

Reply:
yes, and not only that. In the case of failure, it will indicate the exact state at which it had happened. In general, more context is always better.